backend/authapp/utils.py

```diff
@@ -1,12 +1,16 @@
+from utils.general import str2bool
+
 def can_modify(request, instance):
     modifying_user = request.user.username
     modified_user = instance.username
+
     modified_user_is_staff = instance.is_staff
+    modifying_user_is_staff = str2bool(request.data.get('is_staff', 'true'))
 
-    if modified_user == 'admin':
+    if modified_user == 'admin' and modifying_user_is_staff == False:
```

Comment:
Isn't it the case that if the modifying user isn't staff, than this person can only modify their own username? This current conditional seems to be a specific subset of this requirement, because it is just checking if a non-staff user is attempting to modify the `admin` user.